### ISSUE TITLE
the val_loss can't converge in tensorflow 1.15.2,  while tf 1.2.1 works

### ISSUE THREAD

**Issue by Tangzy7** [hugochan/K-Competitive-Autoencoder-for-Text-Analytics#32]
Running train.py with Tensorflow 1.15.2 in 20news, I got the val_loss 0.6x, while the training loss got 0.0x
However, the problem won't happen in tensorflow 1.2.1.   Other dependencies are same with requirement.txt

Would you mind if you check this problem?

**Comment by hugochan** [comment]
> Running train.py with Tensorflow 1.15.2 in 20news, I got the val_loss 0.6x, while the training loss got 0.0x
> However, the problem won't happen in tensorflow 1.2.1. Other dependencies are same with requirement.txt
> 
> Would you mind if you check this problem?

It might be because Tensorflow 1.15.2 made some important changes which actually made some difference here. It's been a long time since the release of TensorFlow 1.2.1. Can you please  share what changes you made to the original code in order to adapt it to TensorFlow 1.15.2? You might also want to check out #24.

**Comment by Tangzy7** [comment]
> > Running train.py with Tensorflow 1.15.2 in 20news, I got the val_loss 0.6x, while the training loss got 0.0x
> > However, the problem won't happen in tensorflow 1.2.1. Other dependencies are same with requirement.txt
> > Would you mind if you check this problem?
> 
> It might be because Tensorflow 1.15.2 made some important changes which actually made some difference here. It's been a long time since the release of TensorFlow 1.2.1. Can you please share what changes you made to the original code in order to adapt it to TensorFlow 1.15.2? You might also want to check out #24.

I just change some codes to adapt python3, while made no changes to the model. I just saw the requirements.txt said to use tensorflow 1.15.2.

**Comment by hugochan** [comment]
> > > Running train.py with Tensorflow 1.15.2 in 20news, I got the val_loss 0.6x, while the training loss got 0.0x
> > > However, the problem won't happen in tensorflow 1.2.1. Other dependencies are same with requirement.txt
> > > Would you mind if you check this problem?
> > 
> > 
> > It might be because Tensorflow 1.15.2 made some important changes which actually made some difference here. It's been a long time since the release of TensorFlow 1.2.1. Can you please share what changes you made to the original code in order to adapt it to TensorFlow 1.15.2? You might also want to check out #24.
> 
> I just change some codes to adapt python3, while made no changes to the model. I just saw the requirements.txt said to use tensorflow 1.15.2.

Hi, the original code was tested under Tensorflow 1.2.1. I changed it to Tensorflow = 1.15.2 in requirements.txt as suggested by Github due to the security issue. Can you try Tensorflow 1.2.1 first to see if it works?
